pkg/cinderbackup/statefulset.go

```diff
@@ -113,12 +111,26 @@ func StatefulSet(
 							SecurityContext: &corev1.SecurityContext{
 								RunAsUser: &runAsUser,
 							},
-							Env:            env.MergeEnvs([]corev1.EnvVar{}, envVars),
-							VolumeMounts:   GetVolumeMounts(),
-							Resources:      instance.Spec.Resources,
-							ReadinessProbe: readinessProbe,
-							LivenessProbe:  livenessProbe,
-							StartupProbe:   startupProbe,
+							Env:           env.MergeEnvs([]corev1.EnvVar{}, envVars),
+							VolumeMounts:  GetVolumeMounts(),
+							Resources:     instance.Spec.Resources,
+							LivenessProbe: livenessProbe,
+							StartupProbe:  startupProbe,
+						},
+						{
+							Name: "probe",
+							Command: []string{
+								"/usr/local/bin/container-scripts/healthcheck.py",
+								"backup",
+								"/var/lib/config-data/merged/cinder.conf",
+							},
+							Image: instance.Spec.ContainerImage,
+							SecurityContext: &corev1.SecurityContext{
+								RunAsUser: &runAsUser,
+							},
+							Env:          finalEnvVars,
+							VolumeMounts: volumeMounts,
+							Resources:    instance.Spec.Resources,
```

Comment:
I'm ambivalent as to whether we should set the probe pod's `Resources` to the value provided in the CR's spec.  Maybe we want to hardcode something here because this probe would be pretty static in its CPU/mem consumption?  Or perhaps leave it out all together and let k8s decide or use a default?